llmfoundry/callbacks/monolithic_ckpt_callback.py

```diff
@@ -74,6 +74,9 @@ def _save_checkpoint(self, state: State, logger: Logger) -> None:
         ) if self.upload_to_object_store else contextlib.nullcontext(
             enter_result=save_dir)
         with dir_context_mgr as temp_save_dir:
+            assert isinstance(temp_save_dir,
+                              str)  # pyright doesn't know about enter_result
```

Comment:
```suggestion
            assert isinstance(temp_save_dir, str)  # pyright typing
```
Nit: i'd like to keep it one-line